projects/igniteui-angular/src/lib/grids/grid/grid.master-detail.spec.ts

```diff
@@ -704,10 +704,12 @@ describe('IgxGrid Master Detail #grid', () => {
 
     describe('Integration', () => {
         describe('Paging', () => {
-            it('Should not take into account expanded detail views as additional records.', fakeAsync(() => {
+            it('Should not take into account expanded detail views as additional records.', async () => {
                 fix = TestBed.createComponent(DefaultGridMasterDetailComponent);
                 grid = fix.componentInstance.grid;
+
                 fix.detectChanges();
+                await wait(); // trigger full ngAfterViewInit cycle to be sure rendered = true
```

Comment:
Wouldn't a `tick()` do the trick as well?
I mean `ngAfterViewInit` is run with the first `fix.detectChanges()` before this, so we're not really waiting on it. Is this about the rendered flag? That's a promise so wait/tick would do indeed, but it's somewhat obscure. If you want the event, just wait for it?
```ts
await grid.rendered.toPromise();
```
Then again if that's to get the pipes to populate the `pagingState`, again I'm kind of confused what necessitated this

Reply:
As discussed this is related to enabling paging after the grid is rendered and is related to [this](https://github.com/IgniteUI/igniteui-angular/pull/12229/files#diff-9511c230760b50c458554bf036523d4de868b74f0f9fbfcb80d5428068b938caL75) change, because it initially has null values for `page` and `perPage` and when paginator is added this triggers the paging pipe. For now the pipe is triggered by [this pipeTrigger](https://github.com/IgniteUI/igniteui-angular/pull/12229/files#diff-440a9d274756495c7bb4e7acf64e5ff3757d060999c43e932fe7654b565f793eR3867), but since this is a change that should affect mainly angular elements.

We discussed that maybe best would be to introduce a separate paging pipe trigger when the paginator is added like `pipeTrigger`, to keep the old behavior of the grid.

Reply:
Yes, at this point the best thing to do is revert this change in the template (and related tests) for this PR and leave any changes to https://github.com/IgniteUI/igniteui-angular/pull/12647 since I moved the paging changes there anyway

Reply:
reverted